dump.go

```diff
@@ -110,12 +119,31 @@ func (d *Dumper) DumpDDLs(ctx context.Context) error {
 	}
 
 	for _, ddl := range resp.Statements {
+		if len(d.tables) > 0 && !d.tables[parseTableNameFromDDL(ddl)] {
+			continue
+		}
 		fmt.Fprintf(d.out, "%s;\n", ddl)
 	}
 
 	return nil
 }
 
+func parseTableNameFromDDL(ddl string) string {
+	ddl = strings.ReplaceAll(ddl, "\n", "")
+	if indexRegexp.MatchString(ddl) {
+		match := indexRegexp.FindStringSubmatch(ddl)
+		return match[1]
+	}
+	if tableRegexp.MatchString(ddl) {
+		match := tableRegexp.FindStringSubmatch(ddl)
+		return match[1]
+	}
+	return ""
+}
+
+var indexRegexp = regexp.MustCompile("^\\s*CREATE\\s+(?:UNIQUE\\s+|NULL_FILTERED\\s+)?INDEX\\s+(?:[a-zA-Z0-9_`]+)\\s+ON\\s+`?([a-zA-Z0-9_]+)`?")
+var tableRegexp = regexp.MustCompile("^\\s*CREATE\\s+TABLE\\s+`?([a-zA-Z0-9_]+)`?")
+
```

Comment:
I'm sorry to bother you many times, but I found DDL could have `ALTER TABLE table_name` statement when we use [Foreign Key](https://cloud.google.com/spanner/docs/foreign-keys/how-to) constraints.

For example, This is what I get from Spanner for DDLs containing foreign key.
```
CREATE TABLE t5 (
  Id INT64 NOT NULL,
  T6Id INT64 NOT NULL,
) PRIMARY KEY(Id);

CREATE TABLE t6 (
  Id INT64 NOT NULL,
) PRIMARY KEY(Id);

ALTER TABLE t5 ADD FOREIGN KEY(T6Id) REFERENCES t6(Id);
```

Could you consider supporting this as well?

Reply:
No problem, let me add foreign key support